src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java

```diff
@@ -19,340 +19,114 @@
 
 package com.puppycrawl.tools.checkstyle.api;
 
-import java.util.BitSet;
-
-import antlr.CommonASTWithHiddenTokens;
-import antlr.Token;
-import antlr.collections.AST;
-import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
-
 /**
  * An extension of the CommonAST that records the line and column number.
  *
  * @see <a href="https://www.antlr.org/">ANTLR Website</a>
```

Comment:
please update documentation block, to mention that it is abstraction level to hide details of implementation of AST base on ANTRL.

Reply:
Done.